src/state/ProcessingState.cpp

```diff
@@ -193,7 +193,7 @@ int ProcessingState::runStateProcess() {
     this->logger->log("Loaded image with a width of %d px, a height of %d px and %d channels", width, height, channels);
 
     // data processing
-    int *output = this->segFilter->doProcessing(resizedImage, width, height, channels);
+    int *output = this->segFilter->doProcessing(image, width, height, channels, this->currentModelIndex);
```

Comment:
I think this should be resizedimage instead of image as the image argument (so the correct image size is fed to the model)